pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_hint_handlers.go

```diff
@@ -48,19 +48,42 @@ func (p *DynamicPolicy) sharedCoresHintHandler(ctx context.Context,
 		return nil, fmt.Errorf("got nil request")
 	}
 
-	if !qosutil.AnnotationsIndicateNUMABinding(req.Annotations) {
-		return util.PackResourceHintsResponse(req, string(v1.ResourceCPU),
-			map[string]*pluginapi.ListOfTopologyHints{
-				string(v1.ResourceCPU): nil, // indicates that there is no numa preference
-			})
+	if qosutil.AnnotationsIndicateNUMABinding(req.Annotations) {
+		return p.sharedCoresWithNUMABindingHintHandler(ctx, req)
 	}
 
-	return p.sharedCoresWithNUMABindingHintHandler(ctx, req)
+	// TODO: support sidecar follow main container for normal share cores in future
+	if req.ContainerType == pluginapi.ContainerType_MAIN {
+		ok, err := p.checkNormalShareCoresCpuResource(req)
+		if err != nil {
+			general.Errorf("failed to check share cores cpu resource for pod: %s/%s, container: %s",
+				req.PodNamespace, req.PodName, req.ContainerName)
+			return nil, fmt.Errorf("failed to check share cores cpu resource: %q", err)
+		}
+
+		if !ok {
+			_ = p.emitter.StoreInt64(util.MetricNameShareCoresNoEnoughResourceFailed, 1, metrics.MetricTypeNameCount, metrics.ConvertMapToTags(map[string]string{
+				"resource":      v1.ResourceCPU.String(),
+				"podNamespace":  req.PodNamespace,
+				"podName":       req.PodName,
+				"containerName": req.ContainerName,
+			})...)
+			return nil, fmt.Errorf("no enough cpu resource")
+		}
+	}
+
+	return util.PackResourceHintsResponse(req, string(v1.ResourceCPU),
+		map[string]*pluginapi.ListOfTopologyHints{
+			string(v1.ResourceCPU): nil, // indicates that there is no numa preference
+		})
 }
 
 func (p *DynamicPolicy) reclaimedCoresHintHandler(ctx context.Context,
 	req *pluginapi.ResourceRequest,
 ) (*pluginapi.ResourceHintsResponse, error) {
+	if util.PodInplaceUpdateResizing(req) {
```

Comment:
Both the `PodInplaceUpdateResizing` and `reclaimedCoresHintHandler` methods do not perform a check for `req` being `nil`,  if there might be a risk of panic here.